src/scenic/core/regions.py

```diff
@@ -2055,6 +2062,152 @@ def evaluateInner(self, context):
         )
 
 
+class VoxelRegion(Region):
+    """Region represented by a voxel grid in 3D space.
+
+    Args:
+        encoding: A numpy array encoding the voxel grid.
+        dimensions: An optional 3-tuple, with the values representing width, length, height
+          respectively. The voxel region will be scaled to have these dimensions. Default
+          value (1,1,1).
+        position: A position, which determines where the center of the region will be. Default
+          value is the origin, (0,0,0).
+        voxelGrid: Optionally, just pass in the final trimesh voxelGrid to be used. Passing this
+          parameter will result in encoding, dimensions, and position being ignored.
+        orientation: An optional vector field describing the preferred orientation at every point in
+          the region.
+        name: An optional name to help with debugging.
+        lazy: Whether or not to be lazy about pre-computing internal values. Set this to True if this
+          VoxelRegion is unlikely to be used outside of an intermediate step in compiling/pruning.
+    """
+
+    def __init__(self, voxelGrid, orientation=None, name=None, lazy=False):
+        # Initialize superclass
+        super().__init__(name, orientation=orientation)
+
+        # Work around Trimesh caching bug
```

Comment:
Would be good to put a link to the corresponding trimesh issue to help us remember to remove this when the bug is fixed.

Reply:
Actually the issue has since been resolved but is only available in Trimesh >= 4.0. I'll remove the workaround since we're requiring that in this PR: https://github.com/BerkeleyLearnVerify/Scenic/pull/169

Reply:
Actually, I'll leave in the workaround and remove it once we require a high enough Trimesh version (to avoid CI failures)